packages/sidecar/src/utils/docker-compose.ts

```diff
@@ -0,0 +1,199 @@
+  const pluginVer = composePluginVersion!.replace(/^v/, "");
+  const standaloneVer = composeStandaloneVersion!.replace(/^v/, "");
+
+  if (isVersionGreaterOrEqual(standaloneVer, pluginVer)) {
+    return { command: ["docker-compose"], version: composeStandaloneVersion! };
+  }
+
+  return { command: ["docker", "compose"], version: composePluginVersion! };
+}
+
+/**
+ * Generate the override YAML for injecting Spotlight environment variables
+ */
+function generateSpotlightOverrideYaml(serviceNames: string[], port: number): string {
+  const services: Record<string, unknown> = {};
+
+  // We use host.docker.internal to ensure the container can access the host machine
+  // for backend services that need to access the host machine to send events to the Sidecar.
+  const spotlightUrl = `http://${DOCKER_HOST_INTERNAL}:${port}/stream`;
+  const publicSpotlightUrl = `http://localhost:${port}/stream`;
+
+  for (const serviceName of serviceNames) {
+    services[serviceName] = {
+      environment: [
+        `SENTRY_SPOTLIGHT=${spotlightUrl}`,
+        `NEXT_PUBLIC_SENTRY_SPOTLIGHT=${publicSpotlightUrl}`, // This is needed for Next.js
+        "SENTRY_TRACES_SAMPLE_RATE=1",
+      ],
+      extra_hosts: [`${DOCKER_HOST_INTERNAL}:${DOCKER_HOST_GATEWAY}`],
+    };
+  }
+
+  return stringifyYaml({ services });
+}
+
+/**
+ * Build the Docker Compose command with Spotlight environment injection
+ */
+export function buildDockerComposeCommand(
+  config: DockerComposeConfig,
+  port: number,
+): { cmdArgs: string[]; dockerComposeOverride: string } {
+  const cmdArgs = [...config.command];
+
+  cmdArgs.push("-f", config.composeFile);
+
+  if (config.overrideFile) {
+    cmdArgs.push("-f", config.overrideFile);
+  }
+
+  cmdArgs.push("-f", "-");
+
+  cmdArgs.push("up");
+
+  const dockerComposeOverride = generateSpotlightOverrideYaml(config.serviceNames, port);
+
+  return { cmdArgs, dockerComposeOverride };
+}
+
+/**
+ * Detect and configure Docker Compose
+ *
+ * This function orchestrates the detection of Docker, Docker Compose,
+ * compose files, and services to build a complete configuration.
+ */
+export function detectDockerCompose(): DockerComposeConfig | null {
+  const docker = detectDocker();
+  if (!docker) {
+    logger.debug("Docker is not installed or not in PATH");
+    return null;
+  }
+
+  if (!docker.valid) {
+    return null;
+  }
+
+  logger.debug(`Detected Docker version ${docker.version}`);
+
+  const compose = detectComposeCommand();
+  if (!compose) {
+    logger.debug("Docker Compose is not installed or not in PATH");
+    return null;
+  }
+
+  logger.debug(`Detected Docker Compose version ${compose.version} (${compose.command.join(" ")})`);
+
+  const composeFile = findComposeFile();
+  if (!composeFile) {
+    logger.debug("No compose file found in current directory");
+    return null;
+  }
+
+  logger.debug(`Found compose file: ${composeFile}`);
+
+  const serviceNames = parseComposeFile(composeFile);
```

Comment:
If we are reading the file right away, we can save one I/O operation (which really counts) by simply trying to read each possible file and if all raises `E_NOENT` we know we don't have the file.